pymeasure/instruments/agilent/agilentE5062A.py

```diff
@@ -45,6 +45,362 @@
 ]
 
 
+class VNATrace(Channel):
+    """A trace (of a channel) of the E5061A/E5062A"""
+
+    placeholder = 'tr'      # default ch would collide with VNAChannel
+
+    def activate(self):
+        """Select this trace. """
+        self.write('CALC{{ch}}:PAR{tr}:SEL')
+
+    parameter = Channel.control(
+        "CALCulate{{ch}}:PARameter{tr}:DEFine?",
+        "CALCulate{{ch}}:PARameter{tr}:DEFine %s",
+        """Control the measurement parameter of the trace (str). Can be
+        S11, S21, S12, or S22""",
+        validator=strict_discrete_set,
+        values=['S11', 'S12', 'S21', 'S22']
+    )
+
+
+class VNAChannel(Channel):
+    """A measurement channel of the E5061A/E5062A"""
+
+    def _add_trace(self, override_id=None):
+        """Internal method that adds a trace to the traces list"""
+        tr_id = len(self.traces) + 1 if override_id is None else override_id
+        self.add_child(VNATrace, tr_id, 'traces', 'tr_')
+
+    def _update_trace_count(self, active_traces):
+        """Internal method that updates the trace list based on the number
+        of active traces.
+        """
+        while len(self.traces) > active_traces:
+            self.remove_child(self.traces[max(self.traces.keys())])
+        while len(self.traces) < active_traces:
+            self._add_trace()
+
+    def _update_power_values(self, attenuation):
+        """Internal method that updates the valid range of stimulus power
+        based on the configured attenuation"""
+        self.power_values = (
+            -attenuation - 5,
+            -attenuation + 10)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # self.traces = []
+        self._add_trace(override_id=1)                # add just a single trace to init it all
+        self._update_trace_count(self.active_traces)  # add remaining traces if necessary
+        self._update_power_values(self.attenuation)   # update dynamic limits of power
+
+    start_frequency = Channel.control(
+        "SENSe{ch}:FREQuency:STARt?", "SENSe{ch}:FREQuency:STARt %g",
+        """Control the start frequency in Hz (float).""",
+        cast=lambda x: int(float(x)),
```

Comment:
I guess, that the response contains something like `1E5`? Or why do you use `int(float(x))` instead of  `cast=int`?

Reply:
int appears to work fine